### PR TITLE
improve docs on modules

### PR DESCRIPTION
**Pull request by stevemao**

### CODE REVIEW

**Comment by stevemao** [comment]
@envision thanks!, I'd reword it to: **NOTE: This will bundle every module in D3**

To try to avoid mentioning tree shaking.

**Comment by milospanasiuk** [comment]
Hello there,

Can someone guide on how to update the wiki with an additional plugin link? Tried various solutions but none was successful. 
When creating a pull request, the wiki is not detected from forked repo since its repo for itself.

**Comment by curran** [comment]
I believe the wiki was locked as a result of abuse/spam. Take a look at the age of the entries there, they are all quite old now.

This may be the closest replacement for listing plugins/packages: https://github.com/wbkd/awesome-d3

**Comment by mbostock** [comment]
Thank you for the suggestion. The behavior of tree-shaking is dependent on the bundler you are using; for example Rollup does tree-shaking fairly intelligently using static analysis. Also, tree-shaking may be affected by adoption of the sideEffects flag #3131 

While I know this PR is well-intentioned, I do not want to take responsibility for documenting how bundlers behave. People should refer to their preferred bundler’s documentation instead.
